tests/chainerx_tests/unit_tests/test_array.py

```diff
@@ -42,6 +44,81 @@ def test_init_with_device(shape, dtype_spec, device):
     _check_array(array, dtype_spec, shape, device=device)
 
 
+@pytest.mark.parametrize('value', [
+    0, 1, -1, 0.1, 0.9, -0.1, -0.9, 1.1, -1.1, 1.9, -
+    1.9, True, False, float('inf'), -float('inf'), float('nan'), -0.0
+])
+@pytest.mark.parametrize('shape', [
+    (), (1,), (1, 1, 1)
+])
+@pytest.mark.parametrize_device(['native:0', 'cuda:0'])
+def test_cast_scalar(device, value, shape, dtype):
+    np_dtype = numpy.dtype(dtype)
+    try:
+        np_value = np_dtype.type(value)
+    except (ValueError, OverflowError):
+        return
+
+    a_np = numpy.asarray([np_value], dtype).reshape(shape)
+    a_chx = chainerx.array(a_np)
+
+    def should_cast_succeed(typ):
+        try:
+            typ(np_value)
+            return True
+        except (ValueError, OverflowError):
+            return False
+
+    # Cast to float
+    if should_cast_succeed(float):
+        assert type(float(a_chx)) is float
+        if math.isnan(float(a_np)):
+            assert math.isnan(float(a_chx))
+        else:
+            assert float(a_np) == float(a_chx)
+    # Cast to int
+    if should_cast_succeed(int):
+        assert type(int(a_chx)) is int
+        assert int(a_np) == int(a_chx)
+    # Cast to bool
+    if should_cast_succeed(bool):
+        assert type(bool(a_chx)) is bool
+        assert bool(a_np) == bool(a_chx)
+
+    # item()
```

Comment:
How about using `chainerx.testing.numpy_chainerx_array_equal` since this is a NumPy compat operation and we'll probably want to use this decorator when adding support for the arguments?

Reply:
What do you mean? We're not comparing arrays.

Reply:
Can you not write the test in this fashion https://github.com/chainer/chainer/blob/master/tests/chainerx_tests/unit_tests/routines_tests/test_math.py#L33 ? Python scalars can apparently also be seen as "array_like" (see https://stackoverflow.com/questions/40378427/numpy-formal-definition-of-array-like-objects).

Reply:
Such generalization would only introduce risks. We can simply separate tests.

Reply:
Okay, let's leave it as is. We discussed offline that this decorator shouldn't be used for non-array types, including Python scalars.